extensions/wikia/CuratedContent/CuratedContentController.class.php

```diff
@@ -483,48 +485,54 @@ public function setData( ) {
+			$this->response->setVal( 'status', $status );
 
+		} else {
+			$this->response->setCode( \Wikia\Service\ForbiddenException::CODE );
+			$this->response->setVal( 'message', 'No permissions to save curated content' );
 		}
+	}
+
+	public function getData( ) {
+		global $wgWikiaCuratedContent, $wgUser;
 
-		$this->response->setVal( 'status', $status );
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
 		// TODO: CONCF-961 Set more restrictive header
 		$this->response->setHeader( 'Access-Control-Allow-Origin', '*' );
-	}
 
-	public function getData( ) {
-		global $wgWikiaCuratedContent;
-		$data = [];
+		if ( $wgUser->isAllowed( 'curatedcontent' ) ) {
+			$data = [];
+			if ( !empty( $wgWikiaCuratedContent ) && is_array( $wgWikiaCuratedContent ) ) {
+				foreach ( $wgWikiaCuratedContent as $section ) {
+					// update information about node type
+					$section['node_type'] = 'section';
 
-		if ( !empty( $wgWikiaCuratedContent ) && is_array( $wgWikiaCuratedContent )  ) {
-			foreach ( $wgWikiaCuratedContent as $section ) {
-				// update information about node type
-				$section['node_type'] = 'section';
+					// rename $section['title'] to $section['label']
+					$section['label'] = $section['title'];
+					unset( $section['title'] );
 
-				// rename $section['title'] to $section['label']
-				$section['label'] = $section['title'];
-				unset( $section['title'] );
+					if ( !empty( $section['label'] ) && empty( $section['featured'] ) ) {
+						// load image for curated content sections (not optional, not featured)
+						$section['image_url'] = CuratedContentHelper::getImageUrl( $section['image_id'] );
+					}
 
-				if ( !empty( $section['label'] ) && empty( $section['featured'] ) ) {
-					// load image for curated content sections (not optional, not featured)
-					$section['image_url'] = CuratedContentHelper::getImageUrl( $section['image_id'] );
-				}
+					foreach ( $section['items'] as $i => $item ) {
+						// load image for all items
+						$section['items'][$i]['image_url'] = CuratedContentHelper::getImageUrl( $item['image_id'] );
 
-				foreach ( $section['items'] as $i => $item ) {
-					// load image for all items
-					$section['items'][$i]['image_url'] = CuratedContentHelper::getImageUrl( $item['image_id'] );
+						// update information about node type
+						$section['items'][$i]['node_type'] = 'item';
+					}
 
-					// update information about node type
-					$section['items'][$i]['node_type'] = 'item';
+					$data[] = $section;
 				}
-
-				$data[] = $section;
 			}
-		}
 
-		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
-		$this->response->setVal( 'data', $data );
-		// TODO: CONCF-961 Set more restrictive header
-		$this->response->setHeader( 'Access-Control-Allow-Origin', '*' );
+			$this->response->setVal( 'data', $data );
+		} else {
+			$this->response->setCode( \Wikia\Service\ForbiddenException::CODE );
+			$this->response->setVal( 'message', 'No permissions to access curated content' );
```

Comment:
Question: Why you're returning message if you aren't reading it on front-end side?